app/assets/javascripts/components/Nav.js

```diff
@@ -0,0 +1,283 @@
+var Nav = function($el) {
+  this.viewportWidth = 0;
+  this.smallViewport = 720;
+  this.delay = 250; // ms
+  this.$mobileNavButton = $(document).find('[data-mobile-nav-button]');
+  this.$mobileNavOverlay = $(document).find('[data-mobile-nav-overlay]');
+  this.$nav = $el;
+  this.$navLevel_1 = this.$nav.find('[data-nav-level-1]');
+  this.$navLevel_1_item = this.$nav.find('[data-nav-level-1-item]');
+  this.$navLevel_1_Heading = this.$nav.find('[data-nav-level-1-heading]');
+  this.$navLevel_2 = this.$nav.find('[data-nav-level-2]');
+  this.$navLevel_2_Heading = this.$nav.find('[data-nav-level-2-heading]');
+  this.$navLevel_2_Extended_Heading = this.$nav.find('[data-nav-level-2-extended-heading]');
+  this.$navLevel_3 = this.$nav.find('[data-nav-level-3]');
+  this.$navLevel_3_Heading = this.$nav.find('[data-nav-level-3-heading]');
+  this.$mobileNavClose = this.$nav.find('[data-mobile-nav-close]');
+  this.$searchBar = this.$nav.find('[data-nav-search-bar]');
+};
+
+/**
+* Initialize the component
+*/
+Nav.prototype.init = function() {
+  this._bindEvents();
+  this._setUpMobileAnimation();
+  this._setUpMobileInteraction();
+  this._setUpDesktopInteraction();
+};
+
+/**
+ * Set up component
+ */
+Nav.prototype._bindEvents = function() {
+  this.$nav.removeClass('uninitialised');
+  $(window).on('resize', this._debounce($.proxy(this._setUpMobileAnimation, this), 100));
```

Comment:
I have replaced JQuery $.proxy with native bind() though $.debounce is not native to JQuery so I am leaving that as is for the minute. This will be replaced by our Dough utility.